src/main/kotlin/com/frcteam3636/frc2025/subsystems/drivetrain/AbsolutePose.kt

```diff
@@ -1,95 +1,121 @@
 package com.frcteam3636.frc2025.subsystems.drivetrain
 
-import edu.wpi.first.apriltag.AprilTagFieldLayout
-import edu.wpi.first.apriltag.AprilTagFields
+//import org.photonvision.PhotonCamera
+//import org.photonvision.PhotonPoseEstimator
+import com.frcteam3636.frc2025.utils.LimelightHelpers
+import com.frcteam3636.frc2025.utils.QuestNav
 import edu.wpi.first.math.Matrix
 import edu.wpi.first.math.VecBuilder
 import edu.wpi.first.math.estimator.SwerveDrivePoseEstimator
+import edu.wpi.first.math.geometry.Pose2d
 import edu.wpi.first.math.geometry.Pose3d
-import edu.wpi.first.math.geometry.Rotation3d
-import edu.wpi.first.math.geometry.Translation3d
 import edu.wpi.first.math.numbers.N1
 import edu.wpi.first.math.numbers.N3
-import edu.wpi.first.math.util.Units
-import edu.wpi.first.networktables.NetworkTableInstance
-import edu.wpi.first.units.Units.DegreesPerSecond
+import edu.wpi.first.units.Units.Meters
+import edu.wpi.first.units.Units.Seconds
+import edu.wpi.first.units.measure.Time
 import edu.wpi.first.util.struct.Struct
 import edu.wpi.first.util.struct.StructSerializable
+import edu.wpi.first.wpilibj.Alert
+import edu.wpi.first.wpilibj.Alert.AlertType
 import edu.wpi.first.wpilibj.Timer
 import org.littletonrobotics.junction.LogTable
-import org.littletonrobotics.junction.Logger
 import org.littletonrobotics.junction.inputs.LoggableInputs
-//import org.photonvision.PhotonCamera
-//import org.photonvision.PhotonPoseEstimator
+import org.team9432.annotation.Logged
 import java.nio.ByteBuffer
-import kotlin.math.pow
-
-interface AbsolutePoseIO {
-    class Inputs : LoggableInputs {
-        // The most recent measurement from the pose estimator.
-        var measurement: AbsolutePoseMeasurement? = null
 
-        override fun toLog(table: LogTable) {
-//            if (measurement != null) {
-//                table.put("Measurement", measurement)
-//            }
+class AbsolutePoseProviderInputs : LoggableInputs {
+    /**
+     * The most recent measurement from the pose estimator.
+     */
+    var measurement: AbsolutePoseMeasurement? = null
+
+    /**
+     * If there is little to no ambiguity.
+     */
+    var isHighQualityReading = false
+
+    /**
+     * Whether the provider is connected.
+     */
+    var connected = false
+
+    override fun toLog(table: LogTable) {
+        if (measurement != null) {
+            table.put("Measurement", measurement)
         }
+        table.put("Connected", connected)
```

Comment:
Looks like I forgot to log `isHighQualityReading`. Although, I'm not totally sure that's an input because it's calculated.